torch/_prims/__init__.py

```diff
@@ -2467,6 +2473,73 @@ def _full_like_aten(
     doc=_full_like_doc,
 )
 
+
+#
+# Linear algebra (linalg) prims
+#
+
+
+def _svd_meta(
+    A: TensorLikeType, *, full_matrices: bool
+) -> Tuple[TensorLikeType, TensorLikeType, TensorLikeType]:
+    A_shape = A.shape
+    check(
+        len(A_shape) >= 2,
+        lambda: f"linalg.svd: input should have at least 2 dimensions, but has {A.ndim} dimensions instead",
+    )
+    batch = A_shape[:-2]
+    m, n = A_shape[-2:]
+    k = min(m, n)
+
+    def contig_strides(shape, *, row_major):
+        if row_major:
+            return utils.make_contiguous_strides_for(shape)
+        else:  # row major
+            return utils.make_batched_column_major_strides_for(shape)
+
+    shape_U = batch + (m, m if full_matrices else k)
+    strides_U = contig_strides(shape_U, row_major=False)
+    U = TensorMeta(shape=shape_U, strides=strides_U, dtype=A.dtype, device=A.device)
+
+    shape_S = batch + (k,)
+    strides_S = contig_strides(shape_S, row_major=True)
+    S = TensorMeta(
+        shape=shape_S,
+        strides=strides_S,
+        dtype=utils.corresponding_real_dtype(A.dtype),
+        device=A.device,
+    )
+
+    shape_Vh = batch + (n if full_matrices else k, n)
+    # The CPU backend returns V, but the cuSolver backend returns V^H
+    # TODO The MAGMA backend returns V, so this is wrong if used with the MAGMA backend
+    is_cuda = (A.device.type == "cuda")
+    strides_Vh = contig_strides(shape_Vh, row_major=is_cuda)
+    Vh = TensorMeta(shape=shape_Vh, strides=strides_Vh, dtype=A.dtype, device=A.device)
+    return U, S, Vh
+
+
+def _svd_aten(
+    A: TensorLikeType, *, full_matrices: bool
+) -> Tuple[Tensor, Tensor, Tensor]:
```

Comment:
Not sure whether we should model here the `compute_uv` parameter. On the one hand, it is useful when computing `svdvals`, but only when no gradients are required. As such, this optimisation is not fully contemplated at the moment in PrimTorch, given that we are not considering gradients yet.
Should we add the `compute_uv` parameter here (as we do in `_linalg_svd` in PyTorch really)?

Reply:
Good thing we can always change our mind later -- let's try it without `compute_uv` for now